src/query_ctrl.js

```diff
@@ -100,6 +103,19 @@ export class MQEQueryCtrl extends QueryCtrl {
     this.onChangeInternal();
   }
 
+  functionSegmentChanged(segment, index) {
+    var _this3 = this;
```

Comment:
You can also use `let` for variable declaration. `let` prevents hoisting, so variable will block-scoped. See more http://es6-features.org/#BlockScopedVariables